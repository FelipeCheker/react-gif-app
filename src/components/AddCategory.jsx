import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('Star Wars');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const inputValueCleaned = inputValue.trim();
        if(inputValueCleaned.length <= 1) return;

        // setCategories(categories => [ inputValue, ...categories ]);
        onNewCategory(inputValueCleaned);
        setInputValue('');
    }

  return (
    <form onSubmit={onSubmit}>
        <input 
            type="text" 
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={ onInputChange }
        />
    </form>
    )
}
