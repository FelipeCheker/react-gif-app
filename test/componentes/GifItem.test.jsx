import { GifItem } from "../../src/components/GifItem"
import { render, screen } from '@testing-library/react';



describe('Test en gifItem', () => { 

    const title = 'Titulo';
    const url = 'https://url.com/feli'
    const alt = 'indicador'

    test('Debe hacer match con el snapshot', () => { 
        const {container} = render( 
            <GifItem
            title = {title}
            url = {url}
            />
        );
        expect (container).toMatchSnapshot();
    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render( 
            <GifItem
            title = {title}
            url = {url}
            />
        );
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
     })

     test('Debe mostrar el titulo en el componente', () => { 
        render( 
            <GifItem
            title = {title}
            url = {url}
            />
        );
        expect(screen.getByText(title)).toBeTruthy();
      })

 })
