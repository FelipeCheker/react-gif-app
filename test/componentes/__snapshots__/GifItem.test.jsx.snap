// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en gifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://url.com/feli"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
